### PR TITLE
Feature/pdf download popover

### PR DESCRIPTION
**Pull request by esotrope**
@aaronsleeper @jessicacolton I introduced your recommended changes for the download button & popover. Additionally I changed the button color. Let me know what you think. I'm pretty sure you'll have suggestions about responsive compatibility and the popover jutting out to the right.

screen shot 2017-07-27 at 10 26 58 pm

*Added the pill header styling that is in the PDF reports

screen shot 2017-07-27 at 10 18 13 pm

### CODE REVIEW

File: webapp/src/main/webapp/src/app/report/report-download.component.html [hunk 1]
```diff
@@ -1,58 +1,45 @@
...
+        </label>
+
+        <label class="ml-xs">
+          <button class="btn btn-info" (click)="form.ngSubmit.emit()">{{'labels.reports.download' | translate}}</button>
+        </label>
+
       </div>
+    </form>
+
+    <div *ngIf="errorMessage" class="alert alert-danger">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report-download.component.html]: Not clear on what to do for error conditions. Will an alert-danger with "Unable to generate report at this time" work?

File: webapp/src/main/webapp/src/styles.less [hunk 2]
```diff
@@ -86,3 +86,14 @@ footer {
...
+  display: inline-block;
+}
+.report-download-popover {
+  display: block;
+  max-width: 440px;
+  width:auto;
+  select {
+    font-weight: normal;
+  }
+}
```
Comment (esotrope) [inline webapp/src/main/webapp/src/styles.less]: @aaronsleeper @jessicacolton styles to make the popover width fit the contents

**Comment by bgsavage** [comment]
I wonder if we should consider moving the advanced filters to a similar popover for A.  consistency and B. save on vertical screen real estate to see real-time changes to the data

File: webapp/src/main/webapp/src/app/report/report-download.component.html (hunk 1 above)
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/report/report-download.component.html]: Really should be *ngIf invalid for this specific reason with the contents pointing to that specific error message in translate.  How many potential errors can you have?

Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report-download.component.html]: do we use form invalid when its an error server response? This is if something goes terribly wrong and the pdf service can't be reached.

Comment (bgsavage) [inline webapp/src/main/webapp/src/app/report/report-download.component.html]: Well my point here is we should separate the concern of whether or not to show the error message from what the error message should be.  Keep the translate in the html when possible and control visibility of the error message with a different boolean.  This is the approach we've been taking in other components until now.  Sorry about the "invalid" terminology confusion.

File: webapp/src/main/webapp/src/styles.less (hunk 2 above)
Comment (aaronsleeper) [inline webapp/src/main/webapp/src/styles.less]: @esotrope was the font bolded previously? i feel like i'm missing context

Comment (esotrope) [inline webapp/src/main/webapp/src/styles.less]: yeah somehow the font weight was bold on the select boxes in the popover.

File: webapp/src/main/webapp/src/app/report/report-download.component.html (hunk 1 above)
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/report-download.component.html]: Yeah totally on board with that - I'll put in a placeholder message

File: webapp/src/main/webapp/src/app/report/student-report-download.component.ts [hunk 3]
```diff
@@ -41,8 +37,8 @@ export class StudentReportDownloadComponent extends ReportDownloadComponent {
           saveAs(download.content, download.name);
         },
         (error: any) => {
-          // TODO: add handler for 404/500 errors
-          this.errorMessage = `${error.status}: ${error.statusText}`;
+          // TODO: add handler for 404/500 errors - consider publishing to global notification "snack drawer"
+          console.error(error);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/report/student-report-download.component.ts]: @wtritch @bgsavage got rid of the error message in the popover. Do we still have some global message service where we can shoot an error message to the user? I'm thinking that might be most appropriate if they async download request fails.

**Comment by esotrope** [comment]
*Merging now
Follow up PR will include suggested vertical popover layout and left/down popover direction.

File: common/src/main/resources/templates/fragments.html [hunk 4]
```diff
@@ -5,11 +5,17 @@
 <body>
 
 <div th:fragment="styles">
-    <link rel="stylesheet"
-          href="https://cdn.rawgit.com/SmarterApp/SBAC-Global-UI-Kit/3f548f92980dbfcc7fe08d78944ea610eaca27d6/dist/css/sbac-ui-kit-wkhtmltopdf.min.css">
+    <link rel="stylesheet" href="https://cdn.rawgit.com/SmarterApp/SBAC-Global-UI-Kit/3f548f92980dbfcc7fe08d78944ea610eaca27d6/dist/css/sbac-ui-kit-wkhtmltopdf.min.css">
```
Comment (esotrope) [inline common/src/main/resources/templates/fragments.html]: @wtritch one thing we will want to add to the docker image is downloading a version of the ui kit from artifactory and using its assets

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -76,6 +77,6 @@ services:
 #      - "dwhost:$DATAWAREHOUSE_HOST"
 
   wkhtmltopdf:
-    image: openlabs/docker-wkhtmltopdf-aas
+    image: fwsbac/wkhtmltopdf-image:latest
```
Comment (esotrope) [inline docker/docker-compose.yml]: now points at will's wkhtmltopdf 0.12.4 image

**Pull request merged**